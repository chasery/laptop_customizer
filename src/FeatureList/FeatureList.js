import React from "react";
import FeatureGroup from "../FeatureGroup/FeatureGroup";
import "./FeatureList.css";

class FeatureList extends React.Component {
  render() {
    const features = Object.keys(this.props.features).map((feature, idx) => {
      const {
        features,
        selected,
        updateFeature,
        USCurrencyFormat,
      } = this.props;
      const featureHash = feature + "-" + idx;

      return (
        <FeatureGroup
          key={featureHash}
          featureOptions={features[feature]}
          feature={feature}
          selected={selected}
          updateFeature={updateFeature}
        />
      );
    });

    return (
      <form className="main__form">
        <h2>Customize your laptop</h2>
        {features}
      </form>
    );
  }
}

FeatureList.defaultProps = {
  features: [],
  selected: [],
};

export default FeatureList;
