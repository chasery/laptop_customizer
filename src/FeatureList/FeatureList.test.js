import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import FeatureList from "./FeatureList";

describe("FeatureList component", () => {
  it("renders without errors", () => {
    const wrapper = shallow(<FeatureList />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
