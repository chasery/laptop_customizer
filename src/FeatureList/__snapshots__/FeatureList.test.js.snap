// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeatureList component renders without errors 1`] = `
<form
  className="main__form"
>
  <h2>
    Customize your laptop
  </h2>
</form>
`;
